bestla/bestla/bestla_utils.h

```diff
@@ -13,7 +13,15 @@
 //  limitations under the License.
 #pragma once
 #include <type_traits>
-#ifdef _OPENMP
+
+#ifdef CLANGTIDY
+#define NO_CLANGTIDY false
+#else
+#define NO_CLANGTIDY true
+#endif
+
+#define BTLA_OPENMP ((_OPENMP) && (NO_CLANGTIDY))
```

Comment:
Why are codes aware of CLANGTIDY? not reasonable. What if we change clang-tidy to other tools?

Reply:
https://github.com/llvm/llvm-project/issues/56460
how about change to sth more general ```CALNG_CHECK``` or ```__has_include(<omp.h>)```?

Reply:
_OPENMP is a common macro definition for GNU and MSVC if OpenMP is found. if Clang has something else you should find its macro for OpenMP and add it here instead of sth for clang-tidy.

Reply:
_OPENMP also works for clang, but in our CI machine we can't install libomp-xx-dev which needs root access

Reply:
how about define a cmake variable ```BTLA_OMP``` and disable it during clang-tidy check

Reply:
Yes, you can add it in cmake. Clang-tidy should be invisible for source code.

Reply:
https://github.com/intel/neural-speed/pull/29/commits/ed615a3827654b23e1fae6d2b9269843b8df5b2d